libs/ui-watt/src/lib/components/input/date-range-input/watt-date-range-input.component.ts

```diff
@@ -0,0 +1,266 @@
+      startDateInputElement,
+      this.onBeforePaste
+    );
+
+    const endDateInputElement = this.endDateInput.nativeElement;
+    const endDateInputMask = this.inputMaskService.mask(
+      this.inputFormat,
+      this.placeholder,
+      endDateInputElement,
+      this.onBeforePaste
+    );
+
+    // Setup and subscribe for input changes
+    this.rangeInputService.init({
+      startInput: {
+        element: startDateInputElement,
+        initialValue: this.initialValue?.start,
+        mask: startDateInputMask,
+      },
+      endInput: {
+        element: endDateInputElement,
+        initialValue: this.initialValue?.end,
+        mask: endDateInputMask,
+      },
+    });
+
+    this.rangeInputService.onInputChanges$
+      ?.pipe(takeUntil(this.destroy$))
+      .subscribe(([start, end]) => {
+        this.markParentControlAsTouched();
+        this.changeParentValue({ start, end });
+      });
+  }
+
+  /**
+   * @ignore
+   */
+  ngOnDestroy() {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  /**
+   * @ignore
+   */
+  writeValue(dateRange: WattDateRange | null): void {
+    if (!this.startDateInput || !this.endDateInput) {
+      this.initialValue = dateRange;
+      return;
+    }
+
+    const inputEvent = new Event('input', { bubbles: true });
+
+    if (dateRange?.start) {
+      this.startDateInput.nativeElement.value = dateRange.start;
+      this.startDateInput.nativeElement.dispatchEvent(inputEvent);
+    }
+
+    if (dateRange?.end) {
+      this.endDateInput.nativeElement.value = dateRange.end;
+      this.endDateInput.nativeElement.dispatchEvent(inputEvent);
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  registerOnChange(onChangeFn: (value: WattDateRange) => void): void {
+    this.changeParentValue = onChangeFn;
+  }
+
+  /**
+   * @ignore
+   */
+  registerOnTouched(onTouchFn: () => void) {
+    this.markParentControlAsTouched = onTouchFn;
+  }
+
+  /**
+   * @ignore
+   */
+  setDisabledState(isDisabled: boolean): void {
+    this.isDisabled = isDisabled;
+    this.changeDetectorRef.detectChanges();
+  }
+
+  /**
+   * @ignore
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  private changeParentValue = (value: WattDateRange): void => {
+    // Intentionally left empty
+  };
+
+  /**
+   * @ignore
+   */
+  private markParentControlAsTouched = (): void => {
+    // Intentionally left empty
+  };
+
+  /**
+   * @ignore
+   */
+  private onBeforePaste(pastedValue: string): string {
+    // Reverse the pasted value, if starts with "year"
+    if (pastedValue.search(/^\d{4}/g) !== -1) {
+      const sepearators = pastedValue.match(/(\D)/);
+      const seperator = sepearators ? sepearators[0] : '';
+      return pastedValue.split(seperator).reverse().join(seperator);
+    }
+    return pastedValue;
+  }
+
+  /**
+   * @ignore
+   */
+  private getInputFormat(): string {
+    const localeDateFormat = getLocaleDateFormat(
+      this.locale,
+      FormatWidth.Short
+    );
```

Comment:
question: why do we need to get the date format from `getLocaleDateFormat` when we change it anyway. Isn't it better to change `FormatWidth.Short` and use that?

Reply:
This is to support both english and danish formats, however this could be removed since we're always using danish date formats atm. in our cases.

Reply:
I think EO are only using `en-us` locale. So if they decide to use the component, the input format and placeholder will look slightly different.

Reply:
If the locale is en-us the placeholder and inputformat should be correct, however the pasted value would be reversed (this is fixed now)